### PR TITLE
[Pretest]: added the proposal for checksum verification

### PR DESCRIPTION
**Pull request by Gmin2**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references 

- [ ] N
- [x] Y 

Related to #394 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [x] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y

### CODE REVIEW

**Comment by Gmin2** [comment]
cc @zong-zhe

**Comment by zong-zhe** [comment]
Hi @Gmin2 😄 

In addition, try to design a separate module called `Checker` that can be used to check whether a KCL dependency is valid. This includes checking whether the dependency's name is appropriate, whether the version number matches the semantic version, and the checksum rather than simply making changes to an existing process. Independent modules can help us better test and manage

**Comment by Gmin2** [comment]
> Hi @Gmin2 😄
> 
> In addition, try to design a separate module called `Checker` that can be used to check whether a KCL dependency is valid. This includes checking whether the dependency's name is appropriate, whether the version number matches the semantic version, and the checksum rather than simply making changes to an existing process. Independent modules can help us better test and manage.

Done @zong-zhe

File: research/checksum-verification-proposal.md [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+- Implement a gradual rollout strategy using the existing FLAG_NO_SUM_CHECK.
+- Generate log warnings for packages without checksums initially.
+- Make checksum verification optional during a transition period.
+- After the transition period, make checksum verification mandatory for all packages.
+
+## Checker Module
+
+To improve modularity and testability, we'll introduce a separate Checker module for validating KCL dependencies. This module will handle name validation, version checking, and checksum verification.
+
+### Checker Module Design
```
Comment (zong-zhe) [inline research/checksum-verification-proposal.md:262]: See https://github.com/kcl-lang/kpm/pull/431#discussion_r1717817452 for more details.

**Pull request closed without merging**